### PR TITLE
Run tests via cli using karma & PhantomJS for Travis

### PR DESCRIPTION
**Pull request by kaeff**
Instead of jasmine's HTML runner, use karma to run specs. karma & all
other dependencies are installed via npm and executed via node.js. This
allows TravisCI to execute the test, and as a side effect, bumps the
versions on the testing toolchain.

- Install node.js
- Run `npm install` once to download dependencies.
- Run `npm test` to run all tests

Things to bear in mind:
- This commit adds general project information in `package.js`
- `karma.conf.js` specifies the order in which src, spec and lib files
  are loaded
- Switch to jasmine spies instead of sinon

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,29 @@
+{
+  "name": "srp_js",
+  "version": "0.0.1",
```
Comment (azul) [inline package.json]: hmm... the Version file says 0.3.2 or so. The Readme states
```
The original repository had code working with a django server. This was removed after the commit tagged 0.1.0
```
The commit history and tags do not show these. No idea why. But maybe we can make this 0.4.0 to make it less confusing and at least clearify this is a later version than what is mentioned in the Readme.

**Comment by azul** [comment]
Looks good. Only thing i noticed was the version.
Cool we can remove jasmine from the repository this way.

**Comment by kaeff** [comment]
Bumped the version. I didn't notice the `Version` file and forgot to ask / just left the default, sorry.

After this is merged I'll rebase it into #6 to prepare it for a merge.

**Comment by azul** [comment]
Cool, thanks @kaeff

**Comment by kaeff** [comment]
Am I just really tired, or is it correct that the PR says 'merged' but I can't find the commits on master?

**Comment by azul** [comment]
@kaeff oh my... I forgot that github is just a autosynced mirror for some of the leap repositories. I guess it has been overwritten by the sync. I will look into this after dinner.

**Pull request merged**